test/buy-tshirt.spec.ts

```diff
@@ -11,36 +11,64 @@ import {
   BankPaymentPage,
 } from '../src/page';
 
-describe('Buy a t-shirt', () => {
-  const emailKey: string = 'aperdomobo@gmail.com';
-  const passwordKey: string = 'WorkshopProtractor';
-  const menuContentPage: MenuContentPage = new MenuContentPage();
-  const productListPage: ProductListPage = new ProductListPage();
-  const productAddedModalPage: ProductAddedModalPage = new ProductAddedModalPage();
-  const orderSummary: OrderSummaryPage = new OrderSummaryPage();
-  const signInCardStep: SignInCardStepPage = new SignInCardStepPage();
-  const addressStepPage: AddressStepPage = new AddressStepPage();
-  const shippingStepPage: ShippingStepPage = new ShippingStepPage();
-  const paymentStepPage: PaymentStepPage = new PaymentStepPage();
-  const bankPaymentPage: BankPaymentPage = new BankPaymentPage();
-
-  it('then should be bought a t-shirt', async () => {
+describe('Given a web clothes store', () => {
+
+  beforeAll(async () => {
     await browser.get('http://automationpractice.com/');
-    await menuContentPage.goToTshirtMenu();
-    await productListPage.addTshirtToCart();
-    await productAddedModalPage.goToShoppingCartSummary();
+  });
+
+  describe('when buy a t-Shirt', () => {
+
+    beforeAll(async () => {
+      const menuContentPage: MenuContentPage = new MenuContentPage();
+      const productListPage: ProductListPage = new ProductListPage();
+      const productAddedModalPage: ProductAddedModalPage = new ProductAddedModalPage();
+      const orderSummary: OrderSummaryPage = new OrderSummaryPage();
+
+      await menuContentPage.goToTshirtMenu();
+      await productListPage.addTshirtToCart();
+      await productAddedModalPage.goToShoppingCartSummary();
+      await orderSummary.goToLoginStep();
+    });
+
+    describe('and login into application', () => {
+
+      beforeAll(async () => {
+        const emailKey: string = 'aperdomobo@gmail.com';
+        const passwordKey: string = 'WorkshopProtractor';
+        const signInCardStep: SignInCardStepPage = new SignInCardStepPage();
+
+        await signInCardStep.sendEmailandPasswKeys(emailKey, passwordKey);
+        await signInCardStep.signIn();
+      });
+
+      describe('and select default address', () => {
```

Comment:
are you selecting a default address?

Reply:
I'm continuing process with default address

Reply:
you should specify that in the description

Reply:
Should be the describe title more clear in that way?

Reply:
> you should specify that in the description

got it!

Reply:
yes